rasterio/_env.pyx

```diff
@@ -59,12 +59,22 @@ def driver_count():
     return GDALGetDriverCount() + OGRGetDriverCount()
 
 
-cpdef get_gdal_config(key):
-    """Get the value of a GDAL configuration option"""
-    key = key.upper().encode('utf-8')
+cpdef get_gdal_config(key, normalize=True):
```

Comment:
Rasterio translates environment variables from `"ON"` to `True`, but that is unwanted behavior when discovering environment variables that may already exist.  Setting `normalize=False` just passes the value straight from `CPLGetConfigOption()`, unless its `NULL` which is still converted to `None`.